WordPress/gutenberg
# [RNMobile][Embed block] Fix URL update when edited after setting a bad URL of a provider

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4002





## Description


One of the uses of the `isEditingURL` value is to control the visibility of the embed bottom sheet, as it matches the expected behavior for this. However, I noticed that this value is also used to trigger a side-effect that replaces the block, producing that a new block is added before the URL was actually edited (more info can be found in [this issue](https://github.com/WordPress/gutenberg/issues/34792#issuecomment-924096891)).

This PR introduces a new value in the local state of the embed block to specifically control the visibility of the bottom sheet and prevent undesired side effects.

## How has this been tested?



1. Add an Embed block.
1. Set a bad URL of any provider (example: https://youtu.be/BAD_URL).
1. Observe that no error message is thrown because we're getting a valid preview response although the content is not.
1. Tap on the pencil icon located in the toolbar to edit the URL.
1. Modify the URL and apply the change by dismissing the bottom sheet.
1. Tap on the pencil icon again.
1. Observe that the URL changed to the new value.

## Screenshots 
N/A

## Types of changes




Bug fix

## Checklist:
